modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/MessageDeliveryTask.java

```diff
@@ -73,7 +73,7 @@ public TaskHint call() throws Exception {
 
             // Get a slot from coordinator.
             Slot currentSlot = requestSlot(storageQueueName);
-
+            currentSlot.setStorageQueue(storageQueue);
```

Comment:
The thing is we only pass the 'storageQueueName' when requesting for a slot. But we need to set the storage queue object itself.